src/Jackalope/Version/VersionHandler.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Jackalope\Version;
+
+use Jackalope\Session;
+use Jackalope\Transport\AddNodeOperation;
+use Jackalope\Transport\WritingInterface;
+use PHPCR\InvalidItemStateException;
+use PHPCR\NodeInterface;
+use PHPCR\PropertyType;
+use PHPCR\UnsupportedRepositoryOperationException;
+use PHPCR\Util\UUIDHelper;
+use PHPCR\Version\VersionException;
+
+class VersionHandler
+{
+    const MIX_VERSIONABLE = 'mix:versionable';
+    const MIX_SIMPLE_VERSIONABLE = 'mix:simpleVersionable';
+
+    /**
+     * @var WritingInterface
+     */
+    private $objectManager;
+
+    /**
+     * @var Session
+     */
+    private $session;
+
+    public function __construct(Session $session)
+    {
+        $this->objectManager = $session->getObjectManager();
+        $this->session = $session;
+    }
+
+    public function addVersionProperties(NodeInterface $node)
+    {
+        $additionalOperations = array();
+
+        if ($node->hasProperty('jcr:isCheckedOut')) {
+            return $additionalOperations;
+        }
+
+        $session = $node->getSession();
+
+        $node->setProperty('jcr:isCheckedOut', true);
+        if (!$node->hasProperty('jcr:uuid')) {
+            $node->setProperty('jcr:uuid', UUIDHelper::generateUUID());
+        }
+
+        $versionStorageNode = $session->getNode('/jcr:system/jcr:versionStorage');
+        $versionHistory = $versionStorageNode->addNode($node->getIdentifier(), 'nt:versionHistory');
+        $versionHistory->setProperty('jcr:uuid', UUIDHelper::generateUUID());
+        $additionalOperations[] = new AddNodeOperation($versionHistory->getPath(), $versionHistory);
+        $versionHistory->setProperty('jcr:versionableUuid', $node->getIdentifier());
+
+        // TODO Set jcr:copiedFrom if needed
+
+        $versionLabels = $versionHistory->addNode('jcr:versionLabels', 'nt:versionLabels');
+        $additionalOperations[] = new AddNodeOperation($versionLabels->getPath(), $versionLabels);
+        $rootVersion = $versionHistory->addNode('jcr:rootVersion', 'nt:version');
+        $rootVersion->setProperty('jcr:uuid', UUIDHelper::generateUUID());
+        $additionalOperations[] = new AddNodeOperation($rootVersion->getPath(), $rootVersion);
+
+        // TODO add frozen node to root version
+
+        $node->setProperty('jcr:versionHistory', $versionHistory);
+        $node->setProperty('jcr:baseVersion', $rootVersion); // TODO set correct base version
+        $node->setProperty('jcr:predecessors', array($rootVersion));
+
+        return $additionalOperations;
+    }
+
+    /**
+     * Performs a checkin for the node on the given path
+     * @param $path
+     * @return string
```

Comment:
please document $path and explain what is returned.

codestyle. empty line before and after the param block.